task.c

```diff
@@ -0,0 +1,97 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "task.h"
+#include "common.h"
+#include "sockets.h"
+
+typedef struct {
+  /* Either ARG_BY_REF or ARG_BY_VAL. */
+  int8_t type;
+  union {
+    unique_id id;
+    struct {
+      ptrdiff_t val_offset;
+      int64_t val_length;
+    } val;
+  };
+} task_arg;
+
+struct task_spec_impl {
+  function_id func_id;
+  int64_t num_args;
+  /* Position of the argument that is currently constructed. */
+  int64_t arg_cursor;
+  int64_t num_returns;
+  int64_t args_value_size;
+  task_arg ids[0];
```

Comment:
It is the size of the data associated with all the arguments that are passed by value combined. I'm documenting it and renaming to by_value_args_num_bytes to clarify.